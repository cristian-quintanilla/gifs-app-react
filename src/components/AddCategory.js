import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');
	
	const handleInputChange = e => {
		setInputValue(e.target.value);
		// console.log('handleInputChange para pruebas');
	}

	const handleSubmit = e => {
		e.preventDefault();
		// console.log('handleSubmit para pruebas');
		
		// Validación
		if (inputValue.trim().length > 2){
			setCategories(categories => [ inputValue, ...categories ]);
			setInputValue('');
		}
	}

	return (
		<form onSubmit={ handleSubmit } >
			<p>{ inputValue }</p>
			<input
				type="text"
				placeholder="Agregar Categoría..."
				className="input--style-6"
				value={ inputValue }
				onChange={ handleInputChange } />
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}