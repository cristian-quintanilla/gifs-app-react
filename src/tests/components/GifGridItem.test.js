import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el Componente <GifGridItem />', () => {

	const title   = 'Soy el título';
	const url     = 'https://localhost/images/image_1.jpg';
	const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

	test('debe de mostrar el Componente correctamente.', () => {
		expect( wrapper ).toMatchSnapshot();
	});

	test('debe de tener un párrafo con el título', () => {
		const p = wrapper.find('p').text().trim();
		expect( p ).toBe( title );
	});

	test('debe de tener la imagen con alt y src', () => {
		const img = wrapper.find('img');
		// console.log(img.prop('src'));
		// console.log(img.props().alt);

		expect( img.prop('src') ).toBe( url );
		expect( img.props().alt ).toBe( title );
	});

	test('debe de tener un div con la clase animate__fadeIn', () => {
		const div = wrapper.find('div');
		const className = div.prop('className');

		// expect( div.hasClass('animate__fadeIn') ).toBe( true );
		expect( className.includes('animate__fadeIn') ).toBe( true );
	});

});