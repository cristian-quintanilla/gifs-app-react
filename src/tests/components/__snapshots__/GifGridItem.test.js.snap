// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <GifGridItem /> Mostrar el Component correctamente. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy el título"
    src="https://localhost/images/image_1.jpg"
  />
  <p>
    Soy el título
  </p>
</div>
`;

exports[`Pruebas en el Componente <GifGridItem /> debe de mostrar el Componente correctamente. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy el título"
    src="https://localhost/images/image_1.jpg"
  />
  <p>
    Soy el título
  </p>
</div>
`;
